From bfaa37138995c8d9cf007a84bb772ba1016a7161 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 16 Sep 2015 16:26:38 +0200
Subject: [PATCH] FsmPrv.h: drop FSM_CONFIG_INLINE_FUNC and use normal inline

* breaks the build with gcc-5 which is using newer C99 inline semantics
  https://gcc.gnu.org/gcc-5/porting_to.html
* fixes:
| /OE/build/owpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/bin/i586-webos-linux/i586-webos-linux-g++  -fPIC  -m32 -march=i586  --sysroot=/OE/build/owpb/webos-ports/tmp-glibc/sysroots/qemux86  -O2 -pipe -g -feliminate-unused-debug-types -fvisibility-inlines-hidden  -DFSM_CONFIG_USE_CUSTOM_ASSERT=0 -DFSM_CONFIG_WEBOS_FEATURES=1 -fPIC -Wl,--no-undefined -O2 -pipe -g -feliminate-unused-debug-types  -O2 -pipe -g -feliminate-unused-debug-types -fvisibility-inlines-hidden -DNDEBUG  -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -shared -Wl,-soname,libPmStateMachineEngine.so.2 -o libPmStateMachineEngine.so.2.0.0 CMakeFiles/PmStateMachineEngine.dir/src/Fsm.c.o CMakeFiles/PmStateMachineEngine.dir/src/FsmDbg.c.o CMakeFiles/PmStateMachineEngine.dir/src/FsmAssert.cpp.o -lPmLogLib
| /OE/build/owpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/libexec/i586-webos-linux/gcc/i586-webos-linux/5.2.0/ld: error: CMakeFiles/PmStateMachineEngine.dir/src/FsmDbg.c.o: multiple definition of 'IsLogLevelEnabled'
| /OE/build/owpb/webos-ports/tmp-glibc/sysroots/x86_64-linux/usr/libexec/i586-webos-linux/gcc/i586-webos-linux/5.2.0/ld: CMakeFiles/PmStateMachineEngine.dir/src/Fsm.c.o: previous definition here
| collect2: error: ld returned 1 exit status
| CMakeFiles/PmStateMachineEngine.dir/build.make:149: recipe for target 'libPmStateMachineEngine.so.2.0.0' failed

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/FsmBuildConfig.h | 12 ------------
 src/FsmDbg.c         |  2 +-
 src/FsmPrv.h         |  2 +-
 3 files changed, 2 insertions(+), 14 deletions(-)

diff --git a/src/FsmBuildConfig.h b/src/FsmBuildConfig.h
index f3fdbaa..8f09960 100644
--- a/src/FsmBuildConfig.h
+++ b/src/FsmBuildConfig.h
@@ -54,18 +54,6 @@ extern "C" {
 
 
 /**
- * Define the appropriate inline attribute for inline functions
- */
-#ifndef FSM_CONFIG_INLINE_FUNC
-    #ifdef __GNUC__
-        #define FSM_CONFIG_INLINE_FUNC  extern __inline
-    #else
-        #define FSM_CONFIG_INLINE_FUNC  extern __inline
-    #endif
-#endif
-
-
-/**
  * FSM_CONFIG_WEBOS_FEATURES: Controls inclusion of 
  * WEBOS-specific features 
  *  
diff --git a/src/FsmDbg.c b/src/FsmDbg.c
index da2f205..27fe11a 100644
--- a/src/FsmDbg.c
+++ b/src/FsmDbg.c
@@ -65,7 +65,7 @@ ResetLoggingOptions(FsmMachineImpl* pFsm);
  * @param pFsm 
  * @param logOptions 
  * 
- * @return FSM_CONFIG_INLINE_FUNC void 
+ * @return void
  */
 static void
 ApplyLoggingOptions(FsmMachineImpl* pFsm, unsigned int logOptions);
diff --git a/src/FsmPrv.h b/src/FsmPrv.h
index 06522d6..29e4ee3 100644
--- a/src/FsmPrv.h
+++ b/src/FsmPrv.h
@@ -187,7 +187,7 @@ RootStateHandler(FsmState* pState, FsmMachine* pFsm,
                  const FsmEvent* pEvt);
 
 
-FSM_CONFIG_INLINE_FUNC int
+inline int
 IsLogLevelEnabled(const FsmMachineImpl* const pImpl,
                   enum FsmDbgLogLevel   const fsmloglevel,
                   int                   const pmloglevel)
-- 
2.5.1

